waspc/src/Wasp/AppSpec/App/Auth/AuthMethods.hs

```diff
@@ -0,0 +1,105 @@
+{-# LANGUAGE DeriveDataTypeable #-}
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE TemplateHaskell #-}
+
+module Wasp.AppSpec.App.Auth.AuthMethods
+  ( generateAuthMethods,
+    userSignupFieldsForEmailAuth,
+    userSignupFieldsForUsernameAuth,
+    userSignupFieldsForExternalAuth,
+    isAuthMethodExternal,
+    AuthMethod (..),
+    UsernameAndPasswordConfig (..),
+    ExternalAuthConfig (..),
+    EmailAuthConfig (..),
+  )
+where
+
+import Data.Data (Data)
+import Language.Haskell.TH
```

Comment:
Unqalified, unlisted import -> not a good practice!
I would suggest importing it qualified as TH.